test/lit.cfg

```diff
@@ -930,39 +930,68 @@ elif run_os in ['linux-gnu', 'linux-gnueabihf', 'freebsd', 'windows-cygnus', 'wi
     config.target_ld = "ld -L%r" % (make_path(test_resource_dir, config.target_sdk_name))
-elif run_os == 'linux-androideabi':
+elif run_os == 'linux-androideabi' or run_os == 'linux-android':
+    def get_architecture_value(**kwargs):
+        result = kwargs[run_cpu]
+        if result is None:
+          if run_cpu == "armv7s" or run_cpu == "armv7k":
+            result = kwargs["armv7"]
+          elif run_cpu == "arm64":
+            result = kwards["aarch64"]
+        return result
+
+    ndk_platform_tuple = get_architecture_value(armv7="armeabi-v7a",
+                                                aarch64="arm64-v8a")
+    ndk_platform_triple = get_architecture_value(armv7="arm-linux-androideabi",
+                                                 aarch64="aarch64-linux-android")
+    toolchain_directory_name = "{}-{}".format(ndk_platform_triple, config.android_ndk_gcc_version)
+    tools_directory = make_path(config.android_ndk_path, "toolchains",
+        toolchain_directory_name, "prebuilt", "linux-x86_64",
+        ndk_platform_triple, "bin")
     lit_config.note("Testing Android " + config.variant_triple)
     config.target_object_format = "elf"
     config.target_shared_library_prefix = 'lib'
     config.target_shared_library_suffix = ".so"
+    config.target_swiftmodule_name = get_architecture_value(armv7="arm.swiftmodule",
+                                                            aarch64="arm64.swiftmodule")
+    config.target_swiftdoc_name = get_architecture_value(armv7="arm.swiftdoc",
+                                                         aarch64="arm64.swiftdoc")
     config.target_runtime = "native"
     config.target_swift_autolink_extract = inferSwiftBinary("swift-autolink-extract")
     config.target_sdk_name = "android"
-    android_linker_opt = "-L {libcxx} -L {libgcc}".format(
+    android_link_paths_opt = "-L {libcxx} -L {libgcc}".format(
         libcxx=make_path(config.android_ndk_path,
-                         "sources", "cxx-stl", "llvm-libc++", "libs",
-                         "armeabi-v7a"),
+                         "sources", "cxx-stl", "llvm-libc++", "libs", ndk_platform_tuple),
         libgcc=make_path(config.android_ndk_path,
-                         "toolchains",
-                         "arm-linux-androideabi-{}".format(config.android_ndk_gcc_version),
-                         "prebuilt", "linux-x86_64", "lib", "gcc",
-                         "arm-linux-androideabi",
+                         "toolchains", toolchain_directory_name, "prebuilt",
+                         "linux-x86_64", "lib", "gcc", ndk_platform_triple,
                          "{}.x".format(config.android_ndk_gcc_version)))
+    # Since NDK r14 the headers are unified under $NDK_PATH/sysroot, so the -sdk
+    # switch is not enough. Additionally we have to include both the unified
+    # sysroot, and the architecture sysroot.
+    android_include_paths_opt = "-I {sysroot} -I {sysroot_arch}".format(
```

Comment:
I think that we should use `-isystem` here instead as we do not want to see any warnings from the NDK, its the system.